### DISCUSSION TITLE
Roadmap for the Huawei Solar PEES project

### DISCUSSION THREAD

**Discussion by JensenNick** [JensenNick/huawei_solar_pees#51] [Announcements]
Below an unprioritized list of improvements that contributors and I will look into and try to implement. Feel free to suggest other improvements.
- [x] Add support for **MB0 inverters** ~incl. new efficiency calculations~.
- [x] Add support for **MAP0 inverters** ~incl. new efficiency calculations~.
- [ ] Add efficiency calculations for LC0, MAP0 and MB0.
- [ ] ~Add support for the **LUNA 7/14/21-S1 batteries**.~
- [x] Add support for related **iStore products** ~(maybe as a separate branch)~.
- [ ] Add support for the **Smart Charger**.
- [ ] ~Set **initial values** for the efficiency calculation based on choice of inverter model.~
- [x] Auto detect **inverter model**.
- [ ] Incorporate `sensor.inverter_model`
- [ ] ~Auto detect **battery model and size**.~
- [ ] Drop down (with search functionality) menu to choose **electricity price sensors**.
- [ ] Fetch and implement relevant **HA user settings** such as i.e. currency.
- [ ] Upgrade the package to an **integration**.
- [ ] Incorporate a **period selector** to display energy and economy data based on calendar selection.
- [ ] **Debugging Dashboard**
- [ ] Enhanced Top Level **Power/Energy Flow Dashboard**
- [ ] Add support for 3 inverter setup (will require user to volunteer as test user).
- [ ] Add support for setup with 2 x BMS (will require user to volunteer as test user).

**Comment by Roving-Ronin** [comment]
@JensenNick sorry busy recently, and procrastinating from something I need to do atm, but looking at dashboard and thinking... is is possible to get PEES (at least the last column with Yield / Consumption, Economy, NRI, Expenses & Income) able to work with an Energy Period Selector, such as: https://github.com/flixlix/energy-period-selector-plus
Currently PEES appears to show calculations 'from date of install' onwards, being able to select time periods (if possible) would be beneficial.

**Reply by JensenNick** [reply]
@Roving-Ronin PEES include utility meters which will provide you with Hourly, Daily, Monthly, Quarterly and Yearly data. I know it's not the same. I'll ad it to the list above. I don't know how / if it could be done but have had the same thought. I was hoping to get some time to give my own dashboards some "love". They are rater mediocre...

**Comment by MrErichsen** [comment]
Hi @JensenNick, I am not proficient in HA, so am wondering if your solution here can help minimize export when the grid has an overproduction/prices are negative?

**Reply by JensenNick** [reply]
Not in the PEES package. What you should have a look at, is my EFLR package (albeit it is under revision).

**Comment by kiwipaulrob** [comment]
Thanks for all your hard work and the integration is working well. As I don't have batteries I had to customise the code. Would it be possible to have a set of sensors available without battery sensors to make that easier?

**Reply by JensenNick** [reply]
I'm sorry but I will not be able to support parallel codes. The PEES package should work without the need for customization though. I don't know what the issue you've had is, but I can imagine that maybe I need to let the related sensors default to zero if no battery is detected.

**Comment by kiwipaulrob** [comment]
Also, my installation has an efficiency sensor available so could that be integrated in the future?

**Reply by JensenNick** [reply]
See my reply here https://github.com/JensenNick/huawei_solar_pees/discussions/92#discussioncomment-13462252
